Mutagen.Bethesda.Analyzers.Api/IAnalyzer.cs

```diff
@@ -0,0 +1,206 @@
+
+        // Isolated
+        var isolatedParams = new IsolatedDriverParams(
+            mod.ToUntypedImmutableLinkCache(),
+            reportDropbox,
+            mod,
+            new ModPath(mod.ModKey, ""),
+            CancellationToken.None);
+
+        var isolated = Task.WhenAll(IsolatedModDrivers.Drivers
+            .Where(d => d.Applicable)
+            .Select(d => d.Drive(isolatedParams)));
+
+        // Contextual
+        var contextualParams = new ContextualDriverParams(
+            _gameEnvironment.LinkCache,
+            _gameEnvironment.LoadOrder,
+            reportDropbox,
+            CancellationToken.None);
+
+        var contextual = Task.WhenAll(ContextualModDrivers.Drivers
+            .Where(d => d.Applicable)
+            .Select(d => d.Drive(contextualParams)));
+
+        await Task.WhenAll(isolated, contextual);
+
+        // Collect results
+        return collectionReportHandler.Results;
+    }
+
+    public async Task<IEnumerable<AnalyzerResult>> Analyze<TMajorRecord>(TMajorRecord record)
+        where TMajorRecord : IMajorRecordGetter
+    {
+        var collectionReportHandler = new CollectionReportHandler();
+        var reportDropbox = GetReportDropbox(collectionReportHandler);
+        var consumer = _container.Resolve<IWorkConsumer>();
+
+        consumer.Start();
+
+        // Isolated
+        var isolatedParams = new IsolatedRecordAnalyzerParams<TMajorRecord>(
+            record.FormKey.ModKey,
+            record,
+            new ReportContextParameters(_gameEnvironment.LinkCache),
+            reportDropbox);
+
+        var isolatedAnalyzerProvider = _container.Resolve<IAnalyzerProvider<IIsolatedRecordAnalyzer<TMajorRecord>>>();
+        var isolated = Task.WhenAll(isolatedAnalyzerProvider.GetAnalyzers().Select(analyzer =>
+        {
+            return _workDropoff.EnqueueAndWait(() =>
+            {
+                analyzer.AnalyzeRecord(isolatedParams with
+                {
+                    AnalyzerType = analyzer.GetType()
+                });
+            }, CancellationToken.None);
+        }));
+
+        // Contextual
+        var contextualParams = new ContextualRecordAnalyzerParams<TMajorRecord>(
+            _gameEnvironment.LinkCache,
+            _gameEnvironment.LoadOrder,
+            record.FormKey.ModKey,
+            record,
+            reportDropbox);
+
+        var contextualAnalyzerProvider = _container.Resolve<IAnalyzerProvider<IContextualRecordAnalyzer<TMajorRecord>>>();
+        var contextual = Task.WhenAll(contextualAnalyzerProvider.GetAnalyzers().Select(analyzer =>
+        {
+            return _workDropoff.EnqueueAndWait(() =>
+            {
+                analyzer.AnalyzeRecord(contextualParams with
+                {
+                    AnalyzerType = analyzer.GetType()
+                });
+            }, CancellationToken.None);
+        }));
+
+        await Task.WhenAll(isolated, contextual);
+
+        return collectionReportHandler.Results;
+    }
```

Comment:
This feels like it's doing too much and is in the weeds on details it shouldnt be concerned with so directly.   We should evolve our engine so we can just call it here, rather than having what is essentially the "glue" class being so directly involved in how to actually execute the logic.

We could instead improve ContextualEngine to take in a single record if that's of interest, rather than having mostly duplicate logic from ContextualEngine copy pasted here

Reply:
Problem is that something like `ByGenericTypeRecordContextualDriver` is specifically designed to work on a whole load order. So we probably need to have some more alternative options there too, or do you have a different design in mind
If you want to play around with this a little yourself, feel free to do that :)